### PR TITLE
Fix ISO-2022-JP-MS encoding convert

### PR DESCRIPTION
**Pull request by shirosaki**
ISO-2022-JP-MS characters are removed by iconv on sending mail.
Use mbstring functions on sending.
A message with ISO-2022-JP-MS characters currupts by wordwrap.
Use ISO-2022-JP-MS charset in wordwrap.

### CODE REVIEW

**Comment by alecpl** [comment]
Is it a PHP bug? Could you provide some samples? Create some unit tests?

**Comment by shirosaki** [comment]
This is not a PHP bug.
Japanese OS environment would be required to reproduce this issue.

Related to this issue.
https://github.com/roundcube/roundcubemail/pull/5668

Use this plugin.
https://github.com/joniburn/sendcharset/blob/master/sendcharset.php

Change send charset to ISO-2022-JP.

If I send a message with subject, body or to/cc address name that contain Japanese characters including ISO-2022-JP-MS characters like '①②③', Japanese characters are removed or corrupted.

I'll try to add unit tests.

**Comment by shirosaki** [comment]
I've split patches and add a test for each change.

File: program/include/rcmail_sendmail.php [hunk 1]
```diff
@@ -702,7 +706,21 @@ public function email_input_format($mailto, $count = false, $check = true)
...
+                    $head_encoding_mode = 'B';
+                    $name = rcube_charset::convert($name, RCUBE_CHARSET, $charset);
+                    $mb_charset = $charset;
+                    if (rcube_charset::$mb_aliases[$charset]) {
+                        $mb_charset = rcube_charset::$mb_aliases[$charset];
+                    }
+                    mb_internal_encoding($mb_charset);
+                    $name = preg_replace('/^"(.*)"$/', '$1', $name);
+                    $name = mb_encode_mimeheader($name, $mb_charset, $head_encoding_mode, "\r\n", 8);
+                    mb_internal_encoding(RCUBE_CHARSET);
```
Comment (shirosaki) [inline program/include/rcmail_sendmail.php]: This part could be replaced by `Mail_mimePart::encodeMB()` if 
https://github.com/pear/Mail_Mime/pull/18 was accepted.

**Comment by alecpl** [comment]
Whay don't you just set the sendcharset to ISO-2022-JP-MS?

**Comment by shirosaki** [comment]
On Wed, Dec 4, 2019 at 20:16 Aleksander Machniak <notifications@github.com>
wrote:

> Whay don't you just set the sendcharset to ISO-2022-JP-MS
>

Mailer such as Thunderbird cannot decode Base64 header string contains
=?ISO-2022-JP-MS?B? .
ISO-2022-JP-MS looks like PHP specific.

> --
Hiroshi Shirosaki

**Comment by shirosaki** [comment]
@alecpl We have upgraded our roundcube to 1.4 with the patches. It works fine.
Other mailers and web mail services support this. So same behavior is expected in Japan.

This company seems to fix some Japanese encoding problems for enterprise support.
https://www.designet.co.jp/system/support/maintenance/roundcube.html

**Comment by sanmai** [comment]
[See my comment here.](https://github.com/pear/Mail_Mime/pull/18#issuecomment-580528487) 

@shirosaki Am I passing your point along all right?

**Comment by shirosaki** [comment]
@sanmai You are right.
We have to use iso-2022-jp-ms for iso-2022-jp variant characters.

**Pull request merged**